app/validation/validator.py

```diff
@@ -83,6 +85,30 @@ def _validate_json_against_schema(self, json_to_validate):
         except SchemaError as e:
             return '{}'.format(e)
 
+    def _validate_schema_contain_metadata(self, schema):
+
+        errors = []
+
+        # user_id and period_id required downstream for receipting
+        # ru_name required for template rendering
+        default_metadata = ['user_id', 'period_id']
+        schema_metadata = schema['metadata']
+
+        # Find all words that precede with "metadata['" or "meta': '"
+        parsed_metadata = re.findall(r"((?<=metadata\[\')\w+|(?<=meta\': \')\w+)", str(schema))
```

Comment:
Correct me if I'm wrong, but I don't think this regex will identify uses of metadata where the dot notation is used? Currently Author does not use the square bracket notation when piping answers (we tried it and had issues with encoding and having to resort to backspacing quotes in our JSON output that we'll need to resolve).

For now its not a big deal as we don't pipe metadata currently, but in the interim, might it be worth extending this regex to check for `metadata.key` in addition to `metadata['key']`?

Reply:
Yeah this was changed due to runner referencing answer IDs instead of answer aliases when piping.

There was a desire to keep the piping consistent. I don't see any issue with allowing either form, it's a valid way to construct the jinja template expression. And especially not if author's not in a place to adopt square brackets.

Reply:
I made it only bracket notation as that what I was told we would use. Both notations are valid of course, it was just from a point of consistency. If Author requires dot notation thats fine, I'm happy to modify the regex.

Reply:
For now author requires dot notation. Hopefully we'll address that at some point, but it's low down on our priorities at the moment. BTW, raised a PR on Publisher to add in the metadata for these values.

Reply:
Great, thanks.